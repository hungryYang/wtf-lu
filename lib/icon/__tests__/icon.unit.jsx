import Icon from '../icon';
import React from 'react';
import renderer from 'react-test-renderer';
import {mount} from 'enzyme';

describe('Icon测试', () => {
  it('测试Icon', () => {
    const tree = renderer
      .create(<Icon name="add"/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('测试Icon点击', () => {
    const fn = jest.fn()
    const wrapper = mount(<Icon name="add" onClick={fn}/>);
    wrapper.find('svg').simulate('click')
    expect(fn).toBeCalled();
  });
});
