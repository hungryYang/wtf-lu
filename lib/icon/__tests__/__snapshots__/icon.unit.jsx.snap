// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon测试 测试Icon 1`] = `
<svg
  className="wtf-icon"
>
  <use
    xlinkHref="#add"
  />
</svg>
`;
